### ISSUE TITLE
Twitter `UserTweetsAndReplies` API may return deleted Tweets

### ISSUE THREAD

**Issue by 734F96** [JustAnotherArchivist/snscrape#575]
Hello !
First of all, thank you very much for SNScrape, it's an awesome projet.
Second, I'm sorry for my english. It's not my native language.

I'm using SNScrape in my Python code, but to simplify, I reproduced this issue in the CLI.
When scraping some users with `TwitterProfileScraper`, the `UserTweetsAndReplies` API may return deleted Tweet, causing SNScrape to crash with the following exception :
```
Traceback (most recent call last):
  File "C:\...\snscrape\_cli.py", line 116, in _dump_locals_on_exception
    yield
  File "C:\...\snscrape\_cli.py", line 318, in main
    for i, item in enumerate(scraper.get_items(), start = 1):
  File "C:\...\snscrape\modules\twitter.py", line 1592, in get_items
    yield from self._graphql_timeline_instructions_to_tweets(instructions)
  File "C:\...\snscrape\modules\twitter.py", line 1313, in _graphql_timeline_instructions_to_tweets
    yield self._graphql_timeline_tweet_item_result_to_tweet(entry['content']['itemContent']['tweet_results']['result'])
KeyError: 'result'
```

You can reproduce this exception by executing the following command : `snscrape twitter-profile nicetomeet0308`
In this case, the API returns [Tweet ID 1547690599553241090](https://twitter.com/any/status/1547690599553241090), which has been deleted. Here is the content of the `entry` variable in `_graphql_timeline_instructions_to_tweets()` :
```
{'entryId': 'tweet-1547690599553241090',
 'sortIndex': '1547690599553241090',
 'content': {'entryType': 'TimelineTimelineItem',
  'itemContent': {'itemType': 'TimelineTweet',
   'tweet_results': {},
   'tweetDisplayType': 'Tweet'}}
```
As you can see, `tweet_results` is empty.

I tested this issue on a Windows 10 machine, and a Ubuntu 22.04 server, both with Python 3.10 and SNScrape 0.4.3.20220107.dev65+g46a6030 (Latest commit on the `master` branch).

Thanks for reading me, and have a nice day !

**Comment by TheTechRobo** [comment]
Can confirm - here's my dump file:

https://transfer.archivete.am/Wmfs3/snscrape_locals_anmm9buj.zst

**Comment by jpenney** [comment]
Made a quick fix locally, and it seems to be working, but I'm not sure if it's the best option.

```diff
diff --git a/snscrape/modules/twitter.py b/snscrape/modules/twitter.py
index 4bf84f1..48f83dc 100644
--- a/snscrape/modules/twitter.py
+++ b/snscrape/modules/twitter.py
@@ -1310,9 +1310,12 @@ class _TwitterAPIScraper(snscrape.base.Scraper):
                        for entry in instruction['entries']:
                                if entry['entryId'].startswith('tweet-'):
                                        if entry['content']['entryType'] == 'TimelineTimelineItem' and entry['content']['itemContent']['itemType'] == 
'TimelineTweet':
-                                               yield self._graphql_timeline_tweet_item_result_to_tweet(entry['content']['itemContent']['tweet_results
']['result'])
+                                               try:
+                                                       yield self._graphql_timeline_tweet_item_result_to_tweet(entry['content']['itemContent']['tweet_results']['result'])
+                                               except KeyError:
+                                                       _logger.warning('Got unrecognised timeline tweet item(s)')
                                        else:
-                                               logger.warning('Got unrecognised timeline tweet item(s)')
+                                               _logger.warning('Got unrecognised timeline tweet item(s)')
                                elif includeConversationThreads and entry['entryId'].startswith('conversationthread-'):  #TODO show more cursor?
                                        for item in entry['content']['items']:
                                                if item['entryId'].startswith(f'{entry["entryId"]}-tweet-'):
```

**Comment by Sailoriae** [comment]
The problem with catching all the `KeyErrors` is that it potentially hides any changes to the API. Whereas here, we know the problem: It's possible that the Tweet has been deleted. So it might be better to just check that `tweet_results` is not empty.

**Comment by 734F96** [comment]
I think it would be better to simply check if `tweet_results` isn't empty. ^^

**Comment by JustAnotherArchivist** [comment]
This should be handled in combination with #392 most likely, i.e. return some sort of tombstone instead of a `Tweet` item. I'll try to get that done soon.

**Comment by jpenney** [comment]
> I think it would be better to simply check if `tweet_results` isn't empty. ^^

I assumed there was a better solution, which is why I just posted a quick workaround as a comment rather than a PR.

**Comment by 734F96** [comment]
Hello,
This issue has been fixed in the following commit : bf0e720b5a767d395541e9e6b5cdb648c1b72e3c
Thank you very much, and have a nice day !
